### PR TITLE
Review code

### PR DESCRIPTION
**Pull request by chinmay-amuratech**

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,26 @@
+c
+n
+c
+@show.screen.seats
+@show.movie
+cap.seats
+cap=@show.screen
```
Comment (vineet-amuratech) [inline .byebug_history]: delete this. make sure such garbage files are not added in the commit.

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -9,8 +9,15 @@
 //
 // Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
 // about supported directives.
-//
+
 //= require jquery
+//= require semantic_ui/semantic_ui
+//= require jquery.validate
+//= require jquery.validate.additional-methods
```
Comment (vineet-amuratech) [inline app/assets/javascripts/application.js]: remove require_tree, if all assets are explicitly required

File: app/assets/javascripts/movies.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (vineet-amuratech) [inline app/assets/javascripts/movies.coffee]: remove coffee files if not used

File: app/assets/stylesheets/scaffolds.scss [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  &:visited {
+    color: #666;
+  }
+
+  &:hover {
+    color: #fff;
+    background-color: #000;
+  }
+}
+
```
Comment (vineet-amuratech) [inline app/assets/stylesheets/scaffolds.scss]: remove scaffolds.scss, it conflicts with bootstrap

File: app/controllers/bookings_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+class BookingsController < ApplicationController
+
+	def index
+		@user = User.find(current_user.id)
```
Comment (vineet-amuratech) [inline app/controllers/bookings_controller.rb]: please explain this !

File: app/assets/stylesheets/scaffolds.scss (hunk 4 above)
Comment (chinmay-amuratech) [inline app/assets/stylesheets/scaffolds.scss]: Do they have same code???

File: app/controllers/bookings_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+				format.html { render :new }
+				format.json { render json: @booking.errors, status: :unprocessable_entity }
+			end
+		end
+	end
+
+	def update
+	end
+
+	def destroy
```
Comment (vineet-amuratech) [inline app/controllers/bookings_controller.rb]: why is this empty, can't bookings be cancelled ?

File: app/controllers/homes_controller.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+class HomesController < ApplicationController
+  before_filter :authenticate_user!
+	def index
+		@movies = Movie.all.select{|movie|  movie.shows.present?}.group_by {|movies| movies.language}
```
Comment (vineet-amuratech) [inline app/controllers/homes_controller.rb]: why don't you iterate over shows here, and fetch the corresponding movies ?

File: app/controllers/homes_controller.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+	end
+
+	def theater_movies
+		@theatre = Theatre.find(params['id'])
```
Comment (vineet-amuratech) [inline app/controllers/homes_controller.rb]: use better naming conventions here

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 9]
```diff
@@ -0,0 +1,28 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  # You should configure your model like this:
+  # devise :omniauthable, omniauth_providers: [:twitter]
+
```
Comment (vineet-amuratech) [inline app/controllers/users/omniauth_callbacks_controller.rb]: are you using omniauth, remove this if not ?

File: app/views/screens/index.json.jbuilder [hunk 10]
```diff
@@ -0,0 +1,4 @@
+json.array!(@screens) do |screen|
+  json.extract! screen, :id, :name, :capacity
+  json.url screen_url(screen, format: :json)
```
Comment (vineet-amuratech) [inline app/views/screens/index.json.jbuilder]: remove the jbuilder files if not used.

File: app/views/seats/_form.html.erb [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+</div>
+<% end %>
+<script>
+$(document).ready(function(){
+  capacity_left = <%= @screen.capacity - @screen.seats.count %>;
+  $(".number").on("keyup", function() {
+    no_of_seats = $(this).val();
+    if(no_of_seats > capacity_left){
+      $(".btn-action").attr("disabled", "disabled");
+      $(".error-action").show();
```
Comment (vineet-amuratech) [inline app/views/seats/_form.html.erb]: this is not ruby, use 'var' when declaring variables.

File: config/routes.rb [hunk 12]
```diff
@@ -1,4 +1,28 @@
...
+  resources :theatres do
+    resources :screens do
+      resources :seats
+      get 'screen_layout', to: 'bookings#screen_layout'
+    end
+  end
+
+  resources :movies, controller: 'movies' do
+    resources :shows, controller: 'shows' do
+      resources :bookings, controller: 'bookings'
```
Comment (vineet-amuratech) [inline config/routes.rb]: when resource name and controller name is the same, controller is not required in options.

File: app/assets/stylesheets/scaffolds.scss (hunk 4 above)
Comment (vineet-amuratech) [inline app/assets/stylesheets/scaffolds.scss]: yes

File: db/seeds.rb [hunk 13]
```diff
@@ -10,9 +10,23 @@
...
 screen4 = Screen.create(name: 'Screen D', capacity: '120', theatre_id: theatre['_id'])
 
-#movie fields(anme, language, genere, duration, movie_format, poster, trailer_url, rating, release_date, theatre_id, booking_id)
-movie1 = Movie.create(name: 'DDLJ', language: 'Hindi', duration: '3.2', rating: '9', release_date: "Thu Feb 11 1992 12:12:12 GMT+0530 (IST)", theatre_id: theatre['_id'])
-movie2 = Movie.create(name: 'Interstellar', language: 'English', duration: '3', rating: '9', release_date: "Fri Feb 11 2014 12:00:00 GMT+0530 (IST)", theatre_id: theatre['_id'])
+#movie fields(name, language, genre, duration, movie_format, !poster, trailer_url, rating, release_date, theatre_id, booking_id)
+movie = Movie.create(name: 'Bajirao Mastani', language: 'Hindi', genre: 'Drama', duration: '3', movie_format: '2D', trailer_url: 'www.youtube1.com', rating: '10', release_date: "Thu Feb 11 1992 12:00:00 GMT+0530 (IST)", theatre_id: theatre['_id'])
+movie = Movie.create(name: 'Chauranga', language: 'Hindi', genre: 'Drama', duration: '2.5', movie_format: '2D', trailer_url: 'www.youtube1.com', rating: '8', release_date: "Thu Feb 12 1992 13:00:00 GMT+0530 (IST)", theatre_id: theatre['_id'])
```
Comment (vineet-amuratech) [inline db/seeds.rb]: use faker here

File: .byebug_history (hunk 1 above)
Comment (chinmay-amuratech) [inline .byebug_history]: Done.

File: app/controllers/homes_controller.rb [hunk 14]
```diff
@@ -0,0 +1,19 @@
+class HomesController < ApplicationController
```
Comment (shaunak-amura) [inline app/controllers/homes_controller.rb]: rename this to HomeController.

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (chinmay-amuratech) [inline app/assets/javascripts/application.js]: Done.

File: app/assets/javascripts/movies.coffee (hunk 3 above)
Comment (chinmay-amuratech) [inline app/assets/javascripts/movies.coffee]: Done.

File: app/assets/stylesheets/scaffolds.scss (hunk 4 above)
Comment (chinmay-amuratech) [inline app/assets/stylesheets/scaffolds.scss]: Done.

File: app/controllers/bookings_controller.rb (hunk 6 above)
Comment (chinmay-amuratech) [inline app/controllers/bookings_controller.rb]: For now, once booking's done, it's done.

File: app/controllers/homes_controller.rb (hunk 8 above)
Comment (chinmay-amuratech) [inline app/controllers/homes_controller.rb]: Done.

File: app/views/screens/index.json.jbuilder (hunk 10 above)
Comment (chinmay-amuratech) [inline app/views/screens/index.json.jbuilder]: Done.

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 9 above)
Comment (chinmay-amuratech) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Done.

File: app/views/seats/_form.html.erb (hunk 11 above)
Comment (chinmay-amuratech) [inline app/views/seats/_form.html.erb]: Done.

File: config/routes.rb (hunk 12 above)
Comment (chinmay-amuratech) [inline config/routes.rb]: Done.

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 9 above)
Comment (vineet-amuratech) [inline app/controllers/users/omniauth_callbacks_controller.rb]: done what..?

File: app/helpers/homes_helper.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+module HomesHelper
```
Comment (vineet-amuratech) [inline app/helpers/homes_helper.rb]: remove this

File: app/controllers/bookings_controller.rb (hunk 6 above)
Comment (vineet-amuratech) [inline app/controllers/bookings_controller.rb]: do the cancel booking form

**Pull request closed without merging**